kubernetes/cmsweb/services/reqmgr2ms-ruleCleaner.yaml

```diff
@@ -0,0 +1,143 @@
+      targetPort: 8244
+      name: ms-rulecleaner
+    - port: 18244
+      targetPort: 18244
+      name: ms-rulecleaner-mon
+---
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: ms-rulecleaner
+  labels:
+    app: ms-rulecleaner
+  namespace: dmwm
+data:
+  setup-certs-and-run.sh: |
+    #!/bin/bash
+    cp /host/etc/grid-security/* /etc/grid-security
+    echo 'INFO Files in /etc/grid-security'
+    ls -lahZ /etc/grid-security
+    su  -c "cd /data && /data/run.sh" --preserve-environment _reqmgr2ms
+---
+kind: Deployment
+apiVersion: apps/v1
+metadata:
+  labels:
+    app: ms-rulecleaner
+  name: ms-rulecleaner
+  namespace: dmwm
+spec:
+  selector:
+    matchLabels:
+      app: ms-rulecleaner
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: ms-rulecleaner
+        env: k8s #k8s#
+      annotations:
+        prometheus.io/scrape: 'true'
+        prometheus.io/port: "18244"
+    spec:
+      # use hostNetwork to allow communication between reqmgr2ms/reqmon/workqueue and couch
+#       hostNetwork: true
+#       dnsPolicy: ClusterFirstWithHostNet
+      securityContext:
+        runAsUser: 0
+        runAsGroup: 0
+      containers:
+      - image: cmssw/reqmgr2ms #imagetag
+        name: ms-rulecleaner
+        lifecycle:
+          postStart:
+            exec:
+              command:
+              - bash
+              - -c
+              - sudo chmod 0777 /data/srv/logs/reqmgr2ms; sudo chown _reqmgr2ms:_reqmgr2ms /data/srv/logs/reqmgr2ms
+        livenessProbe:
+          exec:
+            command:
+            - cmsweb-ping
+            - "--url=http://localhost:8244/ms-rulecleaner/data/status"
+            - "--authz=/etc/hmac/hmac"
+            - -verbose
+            - "0"
+          initialDelaySeconds: 120
+          periodSeconds: 10
+        resources:
+          requests:
+            memory: "256Mi"
+            cpu: "300m"
+          limits:
+            memory: "3Gi"
+            cpu: "1000m"
+        ports:
+        - containerPort: 8244
+          protocol: TCP
+          name: ms-rulecleaner
+        - containerPort: 18244
+          protocol: TCP
+          name: ms-rulecleaner-mon
+        command:
+        - /bin/bash
+        - /opt/setup-certs-and-run/setup-certs-and-run.sh
+        volumeMounts:
+        - name: proxy-secrets
+          mountPath: /etc/proxy
+          readOnly: true
+        - name: secrets
+          mountPath: /etc/secrets
+          readOnly: true
+        - name: robot-secrets
+          mountPath: /etc/robots
+          readOnly: true
+        - name: hmac-secrets
+          mountPath: /etc/hmac
+          readOnly: true
+        - mountPath: /host/etc/grid-security
+          name: etc-grid-security
+          readOnly: true
+        - name: setup-certs-and-run
+          mountPath: /opt/setup-certs-and-run
+#PROD#  - name: logs-cephfs
+#PROD#    mountPath: /data/srv/logs/reqmgr2ms
+        securityContext:
+          privileged: true
+#      initContainers:
```

Comment:
This blocks shows how to start your service, if it needs to wait for another one, e.g. MongoDB, you may uncomment this block and specify which service it should wait for.

Reply:
Thanks again! The `MSRuleCleaner` service is meant to be totally independent locally. Meaning, it does not depend on any other service inside the POD. While it would be good if it could run after the `MSOutput` service since the `MSRuleCleaner` tries to contact it, so we avoid several  void iterations of the service while waiting for the other one  to start. But this is not a must. We have other means to protect ourselves from such situation.